src/plugins/intel_npu/src/compiler/src/driver_compiler_adapter.cpp

```diff
@@ -4,179 +4,81 @@
-        if (ZE_RESULT_SUCCESS != result) {
-            OPENVINO_THROW("L0 zeDriverGetProperties",
-                           " result: ",
-                           ze_result_to_string(result),
-                           ", code 0x",
-                           std::hex,
-                           uint64_t(result));
-        }
-        if (memcmp(&props.uuid, &uuid, sizeof(uuid)) == 0) {
-            _driverHandle = allDrivers[i];
-            break;
-        }
+    ov::SoPtr<intel_npu::IEngineBackend> soPtrBackend = npuBackends->getIEngineBackend();
+    std::shared_ptr<intel_npu::IEngineBackend> iEngineBackend = soPtrBackend._ptr;
+    std::shared_ptr<ZeroEngineBackend> zeroBackend = nullptr;
+    zeroBackend = std::dynamic_pointer_cast<ZeroEngineBackend>(iEngineBackend);
+    if (!zeroBackend) {
+        OPENVINO_THROW("LevelZeroCompilerAdapter init failed to cast zeroBackend, zeroBackend is a nullptr");
     }
 
-    if (_driverHandle == nullptr) {
-        OPENVINO_THROW("LevelZeroCompilerAdapter: Failed to get properties about zeDriver");
-        return;
-    }
+    uint32_t driverExtVersion = zeroBackend->getDriverExtVersion();
 
-    // query the extension properties
-    uint32_t count = 0;
-    result = zeDriverGetExtensionProperties(_driverHandle, &count, nullptr);
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("L0 zeDriverGetExtensionProperties get count",
-                       " result: ",
-                       ze_result_to_string(result),
-                       ", code 0x",
-                       std::hex,
-                       uint64_t(result));
-    }
-    std::vector<ze_driver_extension_properties_t> extProps;
-    extProps.resize(count);
-    result = zeDriverGetExtensionProperties(_driverHandle, &count, extProps.data());
-    if (ZE_RESULT_SUCCESS != result) {
-        OPENVINO_THROW("L0 zeDriverGetExtensionProperties get properties",
-                       " result: ",
-                       ze_result_to_string(result),
-                       ", code 0x",
-                       std::hex,
-                       uint64_t(result));
-    }
-    const char* graphExtName = nullptr;
-    uint32_t targetVersion = 0;
-    for (uint32_t i = 0; i < count; ++i) {
-        auto& property = extProps[i];
-
-        if (strncmp(property.name, ZE_GRAPH_EXT_NAME, strlen(ZE_GRAPH_EXT_NAME)) != 0) {
-            continue;
-        }
-
-        // If the driver version is latest, will just use its name.
-        if (property.version == ZE_GRAPH_EXT_VERSION_CURRENT) {
-            graphExtName = property.name;
-            targetVersion = property.version;
-            break;
-        }
-
-        // Use the latest version supported by the driver.
-        if (property.version > targetVersion) {
-            graphExtName = property.name;
-            targetVersion = property.version;
-        }
-    }
+    ze_context_handle_t zeContext = (ze_context_handle_t)zeroBackend->getContext();
```

Comment:
c-style casts can be replaced with c++ ones.

Reply:
we have fix it with `static_cast` in the latest commit, thanks